pynicotine/gtkgui/dialogs/download.py

```diff
@@ -0,0 +1,680 @@
+                        folder_path,
+                        0,
+                        {},
+                        inconsistent,
+                        username + folder_path
+                    ],
+                    select_row=False
+                )
+                self.initial_selected_iterators.add(parent_iterator)
+
+            iterator = self.tree_view.add_row(
+                [
+                    file_name,
+                    human_size(size),
+                    selected,
+                    username,
+                    folder_path,
+                    size,
+                    file_attributes,
+                    False,
+                    username + file_path
+                ],
+                select_row=False, parent_iterator=parent_iterator
+            )
+            self.num_files[username][folder_path] += 1
+
+            if selected:
+                self.initial_selected_iterators.add(iterator)
+
+                self.total_selected_size += size
+                self.num_selected_files[username][folder_path] += 1
+            else:
+                self.initial_selected_iterators.discard(parent_iterator)
+
+        self.update_title()
+        self.tree_view.expand_all_rows()
+
+        if partial_files:
+            self.set_in_progress()
+
+        self.tree_view.enable_sorting()
+
+        if has_unselected_files:
+            return
+
+        self.unselect_all_button.set_visible(True)
+
+        if not partial_files:
+            self.select_initial_button.set_visible(False)
+
+    def update_title(self):
+
+        num_selected_files = 0
+
+        for folders in self.num_selected_files.values():
+            for count in folders.values():
+                num_selected_files += count
+
+        self.set_title(_("Download %(num_files)s file(s) (%(total_size)s)") % {
+            "num_files": humanize(num_selected_files),
+            "total_size": human_size(self.total_selected_size)
+        })
+
+    def pulse_progress(self, repeat=True):
+
+        if not self.indeterminate_progress:
+            return False
+
+        self.progress_bar.pulse()
+        return repeat
+
+    def set_in_progress(self):
+
+        self.indeterminate_progress = True
+
+        self.progress_bar.get_parent().set_reveal_child(True)
+        self.progress_bar.pulse()
+        GLib.timeout_add(320, self.pulse_progress, False)
+        GLib.timeout_add(1000, self.pulse_progress)
+
+        self.info_bar.set_visible(False)
+
+    def set_finished(self):
+
+        self.indeterminate_progress = False
+
+        self.progress_bar.set_fraction(1.0)
+        self.progress_bar.get_parent().set_reveal_child(False)
+
+        self.info_bar.set_visible(False)
+        self.download_button.set_sensitive(True)
+
+    def set_failed(self):
+
+        if not self.indeterminate_progress:
+            return
+
+        self.set_finished()
+        self.info_bar.show_error_message(
+            _("Failed to request folder contents.")
```

Comment:
> Better error message in info banner

`"Failed to get folder contents from user $(username)s"`

However, a single failure doesn't mean that requests made to other users in the selection might still respond (and they could fail as well)...

`"Failed to get folder contents from $(num_failed_users)i users"`